addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -681,4 +682,74 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
         }).then(reload);
     },
 });
+
+options.registry.WebsiteSaleProduct = options.Class.extend({
+    willStart: async function () {
+        const _super = this._super.bind(this);
+        this.productID = parseInt(this.$target.find('[data-oe-model="product.product"]').data('oe-id'));
+        this.carousel = $('#o-carousel-product');
+        // The editor will try to rollback any changes made to the carousel.
+       // By binding ourselves onto this function we can filter out records 'observed' for mutations.
+       const filterMutationRecordsSuper = this.options.wysiwyg.odooEditor.options.filterMutationRecords;
+       this.options.wysiwyg.odooEditor.filterMutationRecords = (records) => {
+           records = filterMutationRecordsSuper(records);
+           return records.filter((record) => {
+               return record.target.id !== 'o-carousel-product' &&
+                   !(record.target.classList &&
+                       (record.target.classList.contains('o-wsale-carousel-indicators') ||
+                       record.target.classList.contains('o-wsale-carousel-item')))});
+       };
```

Comment:
Hello @odoo/rd-framework-editor 👋 

This PR allows bulk image edition on products.
However since we update the carousel for a live preview the editor registers it and rolls it back when we change the focus to edit another element.

We found this solution which is _really_ hackish.
Is there a proper solution to this problem?

LMK if anything is unclear.
Cheers

Reply:
Wow, thanks a lot for all this information.

It seems like we can definitely make a cleaner version.